tsdb/head.go

```diff
@@ -1321,6 +1322,8 @@ func (h *Head) gc() {
 	symbols := make(map[string]struct{}, len(h.symbols))
 	values := make(map[string]stringset, len(h.values))
 
+	h.symMtx.Lock()
```

Comment:
Is it correct reading `len(h.symbols)` and `len(h.values)` (above) outside of the lock?

Reply:
I double checked and I believe we should take the lock before `make()` symbols and values above.

Reply:
agree